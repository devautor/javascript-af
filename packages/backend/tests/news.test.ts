import { graphqlTestCall } from './utils/gqlTestClient';
import { prisma } from './utils/createPrismaMock';

test('should query', async () => {
  const writerMock = jest.fn(() => ({
    id: 'cjpxpmpgr000c097495f1kz8w',
    username: 'pantharshit00',
    email: 'pantharshit00@gmail.com',
    githubToken: 's',
    profilePic: 'ss',
    createdAt: '2018-12-21T07:25:09.804Z',
    updatedAt: '2018-12-21T07:25:09.804Z',
  }));

  prisma.news = jest.fn(() => ({
    id: 'cjpxpvzai000o0974ydxa80da',
    title: 'Some thing',
    slug: 'some-thing',
    content: 'some mdx thing',
    previewImage: 's',
    isFeatured: false,
    createdAt: '2018-12-21T07:32:22.501Z',
    updatedAt: '2018-12-21T07:32:22.501Z',
    writer: writerMock,
  }));

  const response = await graphqlTestCall(/* GraphQL */ `
  {
  newsItemBySlug(slug:"some-thing"){
    id
    title
    slug
    content
    previewImage
    isFeatured
    writer{
      id
      username
      email
      githubToken
      profilePic
      createdAt
      updatedAt
    }
    createdAt
    updatedAt
  }
}
  `);
  expect(prisma.news).toHaveBeenCalledWith({ slug: 'some-thing' });
  expect(writerMock).toHaveBeenCalled();
  expect(response).toMatchSnapshot();
});
