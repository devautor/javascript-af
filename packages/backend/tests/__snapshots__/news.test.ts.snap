// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should query 1`] = `
Object {
  "data": Object {
    "newsItemBySlug": Object {
      "content": "some mdx thing",
      "createdAt": "2018-12-21T07:32:22.501Z",
      "id": "cjpxpvzai000o0974ydxa80da",
      "isFeatured": false,
      "previewImage": "s",
      "slug": "some-thing",
      "title": "Some thing",
      "updatedAt": "2018-12-21T07:32:22.501Z",
      "writer": Object {
        "createdAt": "2018-12-21T07:25:09.804Z",
        "email": "pantharshit00@gmail.com",
        "githubToken": "s",
        "id": "cjpxpmpgr000c097495f1kz8w",
        "profilePic": "ss",
        "updatedAt": "2018-12-21T07:25:09.804Z",
        "username": "pantharshit00",
      },
    },
  },
}
`;
